src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.c

```diff
@@ -46,80 +77,1172 @@ const ActorInit En_Kakasi_InitVars = {
+ */
+void EnKakasi_CheckPlayerPosition(EnKakasi* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (globalCtx->sceneNum == SCENE_8ITEMSHOP) {
+        player->actor.world.pos.x = -50.0f;
+        player->actor.world.pos.z = 155.0f;
+    } else if (globalCtx->sceneNum == SCENE_TENMON_DAI) {
+        player->actor.world.pos.x = 60.0f;
+        player->actor.world.pos.z = -190.0f;
+    }
+
+    Math_SmoothStepToS(&player->actor.shape.rot.y, (this->actor.yawTowardsPlayer + 0x8000), 5, 1000, 0);
+}
+
+/*
+ * this goes off every frame of dancing the night away, and song teaching
+ * something to do with cutscene camera?
+ */
+void func_8096FAAC(EnKakasi* this, GlobalContext* globalCtx) {
+    if (this->cutsceneCamId != 0) { // SUBCAM_FREE
+        Math_ApproachF(&this->unk214.x, this->unk238.x, 0.4f, 4.0f);
+        Math_ApproachF(&this->unk214.y, this->unk238.y, 0.4f, 4.0f);
+        Math_ApproachF(&this->unk214.z, this->unk238.z, 0.4f, 4.0f);
+
+        Math_ApproachF(&this->unk220.x, this->unk244.x, 0.4f, 4.0f);
+        Math_ApproachF(&this->unk220.y, this->unk244.y, 0.4f, 4.0f);
+        Math_ApproachF(&this->unk220.z, this->unk244.z, 0.4f, 4.0f);
+
+        Math_ApproachF(&this->unk20C, this->unk210, 0.3f, 10.0f);
+
+        func_8016970C(globalCtx, this->cutsceneCamId, &this->unk220, &this->unk214);
+        func_80169940(globalCtx, this->cutsceneCamId, this->unk20C);
+    }
+}
+
+/*
+ * goes off every frame of song teach, but... doing what?
+ */
+void func_8096FBB8(EnKakasi* this, GlobalContext* globalCtx) {
+    if (globalCtx->msgCtx.unk12048 == 0 || globalCtx->msgCtx.unk12048 == 1 || globalCtx->msgCtx.unk12048 == 2 ||
+        globalCtx->msgCtx.unk12048 == 3 || globalCtx->msgCtx.unk12048 == 4) {
+        // why not 0 < x < 4? fewer branches
+        this->unk190++;
+    }
+    if (this->unk190 != 0 && this->animeIndex != ENKAKASI_ANIME_SIDEWAYS_SHAKING) {
+        EnKakasi_SetAnimation(this, ENKAKASI_ANIME_SIDEWAYS_SHAKING);
+    }
+    if (this->unk190 > 8) {
+        this->unk190 = 8;
+    }
+    if (this->unk190 != 0) {
+        Math_ApproachF(&this->skelanime.animPlaybackSpeed, 1.0f, 0.1f, 0.2f);
+        this->actor.shape.rot.y = this->actor.shape.rot.y + 0x800;
+    }
+}
+
+void EnKakasi_InitTimeSkipDialogue(EnKakasi* this) {
+    EnKakasi_SetAnimation(this, ENKAKASI_ANIME_SLOWROLL);
+    this->actionFunc = EnKakasi_TimeSkipDialogue;
+}
+
+void EnKakasi_TimeSkipDialogue(EnKakasi* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (gSaveContext.respawnFlag != -4 && gSaveContext.respawnFlag != -8) {
+        if (gSaveContext.time != 0x4000 && gSaveContext.time != 0xC000 && (gSaveContext.eventInf[1] & 0x80) == 0) {
+            if (this->actor.textId == 0) {
+                // "Oh, yeah!  How was it? It went by in an instant, right?  I'm still full of energy!"
+                this->actor.textId = 0x1653;
+                gSaveContext.weekEventReg[0x53] &= 0xFE; // a lot of flags
```

Comment:
```suggestion
                gSaveContext.weekEventReg[0x53] &= ~1;
```
Is that comment needed?